mantidproject/mantid
# Added Auto cal file generation for ISIS Powder Gem

**Description of work.**
Added create_cal to the ISIS Powder gem scripts, similar to how they are implemented in ISIS powder pearl, this allows automatic generation of a new calibration file using a run. also added the ability to adjust an existing cal file by passing in the path to it **this feature is a temporary fixed and should be removed once the gem IDF is fixed  in #26013

**To test:**
Download these [files](https://github.com/mantidproject/mantid/files/3310685/Gem.zip), updating the paths in the `config.yaml` enable the ISIS data search archive, then run the following script, the focus should not fail.
```
from isis_powder import Gem

config_file_path = r"/path/to/Gem_config_example.yaml"
Gem_example = Gem(config_file=config_file_path,user_name="test")
Gem_example.create_cal(run_number=87618)
Gem_example.create_vanadium(first_cycle_run_no=86578)
Gem_example.focus(run_number=86578,input_mode="Individual")
```
check that the calibration folder contains `cycle_18_4/new_cal_file.cal` after having run the script
then run
```

config_file_path = r"/path/to/Gem_config_example.yaml"
Gem_example = Gem(config_file=config_file_path,user_name="test")
Gem_example.create_cal(run_number=87618,calibration_to_adjust="path to cycle_18_4/new_cal_file.cal)
```
this should not fail
Fixes #23241 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
